eng/pipelines/templates/steps/azd-login.yml

```diff
@@ -1,6 +1,7 @@
 parameters:
   SubscriptionConfiguration: $(sub-config-azure-cloud-test-resources)
   AzdDirectory: ""
+  ServiceConnectionId: "3d79cc98-46f2-428c-bdd5-861414f85602"
```

Comment:
@danieljurek I assume the right thing to do here is to pull this off of `SubscriptionConfiguration` but I don't know enough about what that structure looks like to know if this value there or not.  These needs to be the ID of the service connection that we want to use.

The AzureCLI@2 task seems to do name to id translation (I am guessing) but I haven't looked into how that works yet and if we could support it.  I found this GUID by looking around at some of our internal configuration files.  I won't check this in until you tell me the right way to do it, but trying to get stuff off the ground and taking some shortcuts.

Reply:
This may or may not help:

- [AzureCli task](https://github.com/microsoft/azure-pipelines-tasks/blob/9a01bdd13f107406603afc20bf088ac68a7223c4/Tasks/AzureCLIV2/azureclitask.ts#L198)
- [vsts-task-lib](https://github.com/microsoft/azure-pipelines-task-lib/blob/a56dfecac46753540e79da0d2c93fb41666c0c3e/node/task.ts#L469)

I think environment variables are being seeded -- I'm not sure if that means you definitely an `azdo` task to get everything working e2e.

Reply:
> I think environment variables are being seeded

Yes, it looks like this happens when you have a [task with an input of type `connectedService:AzureRM`](https://devblogs.microsoft.com/devops/public-preview-of-workload-identity-federation-for-azure-pipelines/#custom-azure-tasks-and-extensions).  So I guess we could make this "just work" in the context of an `AzureCLI@2` task (or create our own).

Reply:
Yes see https://github.com/Azure/azure-sdk-for-net/blob/e7d65a13f3f83119ed54e927654e0512d35f087b/sdk/identity/Azure.Identity/samples/OtherCredentialSamples.md#example-of-using-an-azure-pipelines-task 

The AzurePipelineCredential pulls the needed variables that get set in the AzureCLi task.